src/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -2267,7 +2267,15 @@ internal void AdjustScale(int digits, bool fRound)
                 throw new OverflowException(SQLResource.ArithOverflowMessage);
 
             bNewScale = (byte)(lAdjust + _bScale);
-            bNewPrec = (byte)(Math.Min(s_NUMERIC_MAX_PRECISION, Math.Max(1, lAdjust + _bPrec)));
+
+            if (Value == Decimal.Zero && _bScale == 0 && lAdjust > 0)
+            {
+                bNewPrec = bNewScale;
```

Comment:
Also, in some ad-hoc testing I was able to create a SqlDecimal value equal to Decimal.Zero with Scale==0 and Precision greater than 1. So there should be a check for Precision==1 here.